rust/private/utils.bzl

```diff
@@ -252,11 +252,13 @@ def _invalid_chars_in_crate_name(name):
 
     return dict([(c, ()) for c in name.elems() if not (c.isalnum() or c == "_")]).keys()
 
-def crate_name_from_attr(attr):
+def compute_crate_name(attr, label, toolchain):
```

Comment:
I think we can simplify these parameters a little. We use `attr` to read `name` and `crate_name`. However, `name` can also be read from the `label`. And, passing `ctx.attr` around is a bit of an antipattern (unfortunately it happens a lot in `rules_rust`). 

So what I'd do is pass `label`, `toolchain` as mandatory parameters, and `name_override` as an optional parameter defaulting to `None`. Callers of this function having `crate_name` attribute can call `getattr(ctx.attr, "crate_name", None)` and pass result to `compute_crate_name` without too much hassle.

Reply:
Done.